presto-main/src/main/java/com/facebook/presto/operator/scalar/ReflectionParametricScalar.java

```diff
@@ -286,12 +310,29 @@ public static SqlScalarFunction parseDefinition(Class<?> clazz)
         return builder.build();
     }
 
+    private static <T extends Annotation> Optional<T> getAnnotation(Annotation[] annotations, Class<T> annotationClass)
+    {
+        for (Annotation annotation : annotations) {
+            if (annotationClass.isAssignableFrom(annotation.getClass())) {
+                return Optional.of((T) annotation);
+            }
+        }
+        return Optional.empty();
+    }
+
+    private static <T extends Annotation> Optional<T> getAnnotation(Method method, Class<T> annotationClass)
+    {
+        return getAnnotation(method.getDeclaredAnnotations(), annotationClass);
+    }
+
     private static final class ImplementationParser
     {
         private final String functionName;
-        private final boolean nullable;
+        private boolean nullable;
         private final List<Boolean> nullableArguments = new ArrayList<>();
-        private final String returnType;
+        private boolean returnAsParameter;
```

Comment:
can't be as this is not set in constructor but in method called from constructor.
This is not nice but it is hard to restructure this code to have this fields final and have nice looking initialization.

Probably with major rewrite of this class it would be possible. I wan not focusing on that in this PR. I can clean this up but I will leave it until I fix issues in gist of this PR.